src/Core/Grid/Definition/Factory/MerchandiseReturnProductsGridDefinitionFactory.php

```diff
@@ -0,0 +1,255 @@
+        RequestStack $requestStack
+    ) {
+        parent::__construct($hookDispatcher);
+        $this->setMerchandiseReturnId($requestStack);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId()
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName()
+    {
+        return $this->trans('Merchandise return products', [], 'Admin.Orderscustomers.Feature');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns()
+    {
+        $columns = (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('merchandise_return_bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_order_detail',
+                        'extra_fields' => [
+                            'id_customization',
+                        ],
+                    ])
+            )
+            ->add(
+                (new DataColumn('product_reference'))
+                    ->setName($this->trans('Reference', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'product_reference',
+                    ])
+            )
+            ->add(
+                (new DataColumn('product_name'))
+                    ->setName($this->trans('Product name', [], 'Admin.Shopparameters.Feature'))
+                    ->setOptions([
+                        'field' => 'product_name',
+                    ])
+            )
+            ->add(
+                (new DataColumn('quantity'))
+                    ->setName($this->trans('Quantity', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'product_quantity',
+                    ])
+            )
+            ->add(
+                (new MerchandiseReturnCustomizationColumn('customizations'))
+                    ->setName($this->trans('Customizations', [], 'Admin.Global'))
+                    ->setOptions([
+                        'customizations' => 'customizations',
+                    ])
+            )
+            ->add(
+                (new ActionColumn('actions'))
+                    ->setName($this->trans('Actions', [], 'Admin.Global'))
+                    ->setOptions([
+                        'actions' => (new RowActionCollection())
+                            ->add((new SubmitRowAction('delete'))
+                            ->setName($this->trans('Delete', [], 'Admin.Actions'))
+                            ->setIcon('delete')
+                            ->setOptions([
+                                'route' => 'admin_merchandise_returns_delete_product',
+                                'route_param_name' => 'merchandiseReturnId',
+                                'route_param_field' => 'id_order_return',
+                                'extra_route_params' => [
+                                    'merchandiseReturnDetailId' => 'id_order_detail',
```

Comment:
Already said it but I think this could be simplified with `id_order_detail` And I would even go further, since a OrderReturnDetail can have only one OrderDetail the `id_order_return` should be enough to get back the matching OrderDetail and the the associated Customization